infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/logreader/StreamsLogEntryReader.java

```diff
@@ -197,12 +204,17 @@ public LogReplicationEntry read(UUID logEntryRequestId) throws TrimmedException
                 }
 
                 lastOpaqueEntry = txOpaqueStream.next();
+                deltaCounter.ifPresent(Counter::increment);
```

Comment:
I believe this count should be after L.209 only if lastOpaqueEntryValid is true... cause if not, we will be counting transaction entries which are not intended to be replicated. For instance, consider 5 entries in the transaction stream, but only 1 is valid (has streams to replicate).. we would be counting 5 deltas when we should really count 1... or maybe we can count both (in different metrics)... to know which proportion of the transaction stream ends up being required for replication (1 out of 5 in this case).. I think that could be a nice way of analyzing the data available... to see how noisy the tx stream can be with regards to the data to be replicated..

Reply:
Created the one that counts overall and the one that counts only valid ones.